### DISCUSSION TITLE
Switching actions are sometimes delayed and queued

### DISCUSSION THREAD

**Discussion by AMDFreak2006** [fsaris/home-assistant-awox#111] [Q&A]
Sometimes switching actions taken manually in the GUI to the Awox-Lights and Plugs are only executed 20-60 later and then queued.

What I mean with queued is, that every action which was done in this 20-60 seconds of delay is remembered by HA and then executed at once as fast as possible.
For example one clicks on a plug in the HA gui to switch it on, no reaction occurs, one clicks again for example 3 times, still nothing happens. Around 20-60 seconds later, all 4 clicks get executed, resulting in fast switching of the plug.

HAOS 11.1 with HA core 2023.11.2
home-assistant-awox release 0.1.5

**Comment by fsaris** [comment]
The mesh is known for being slow. And in current state it always tries to control each single light instead of using groups to address them. 

This combined with the issues described in #110 makes the component/integration react slow. If you check the logs you probably see a reconnect action happening when the delay is a few seconds. This is most likely due to the fact that the HA Bluetooth connections interfered with the connection setup by this component. The component requires a dedicated connection something that is resolved when #110 is resolved.

The https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub solution is better in handling the actions. But still "slow" when handling multiple actions fast after each other because the mesh just is slow and needs time to process each action to prevent it from skipping actions.
